maven-core/src/main/java/org/apache/maven/project/DefaultProjectDependenciesResolver.java

```diff
@@ -190,8 +191,20 @@ public DependencyResolutionResult resolve( DependencyResolutionRequest request )
             {
                 if ( !child.getRelocations().isEmpty() )
                 {
-                    logger.warn( "The artifact " + child.getRelocations().get( 0 ) + " has been relocated to "
-                        + child.getDependency().getArtifact() );
+                    org.eclipse.aether.artifact.Artifact artifact = child.getRelocations().get( 0 );
+                    StringBuilder message = new StringBuilder();
+                    message.append( "The artifact " ).append( artifact )
```

Comment:
The StringBuilder is redudant because constant string concat will be turned into a StringBuilder internally anyway. SBs are useful for loops only.

Reply:
Your statement is true, however there are two `if`s in the code that would create another `StringBuilder` and I wanted to avoid that

Reply:
Anyway, I changed that to String if that's an issue